src/test/java/com/typee/logic/commands/AddCommandIntegrationTest.java

```diff
@@ -1,35 +1,82 @@
 package com.typee.logic.commands;
 
+import static com.typee.logic.commands.CommandTestUtil.assertCommandFailure;
+import static com.typee.logic.commands.CommandTestUtil.assertCommandSuccess;
+import static com.typee.testutil.TypicalEngagements.getTypicalEngagementList;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
 import com.typee.model.Model;
+import com.typee.model.ModelManager;
+import com.typee.model.UserPrefs;
+import com.typee.model.engagement.Engagement;
+import com.typee.testutil.EngagementBuilder;
 
 /**
  * Contains integration tests (interaction with the Model) for {@code AddCommand}.
  */
 public class AddCommandIntegrationTest {
 
     private Model model;
-    //5/10/19 Duplicate Person. To be resolved in the future.
-    /*
+
     @BeforeEach
     public void setUp() {
-        model = new ModelManager(getTypicalAddressBook(), new UserPrefs());
+        model = new ModelManager(getTypicalEngagementList(), new UserPrefs());
+    }
+
+    @Test
+    public void execute_newAppointment_success() {
+        Engagement validAppointment = new EngagementBuilder().buildAsAppointment();
+
+        Model expectedModel = new ModelManager(model.getEngagementList(), new UserPrefs());
+        expectedModel.addEngagement(validAppointment);
+        expectedModel.saveEngagementList();
+
+        assertCommandSuccess(new AddCommand(validAppointment), model,
+                String.format(AddCommand.MESSAGE_SUCCESS, validAppointment), expectedModel);
+    }
+
+    @Test
+    public void execute_newInterview_success() {
+        Engagement validInterview = new EngagementBuilder().buildAsInterview();
+
+        Model expectedModel = new ModelManager(model.getEngagementList(), new UserPrefs());
+        expectedModel.addEngagement(validInterview);
+        expectedModel.saveEngagementList();
+
+        assertCommandSuccess(new AddCommand(validInterview), model,
+                String.format(AddCommand.MESSAGE_SUCCESS, validInterview), expectedModel);
     }
 
     @Test
-    public void execute_newPerson_success() {
-        Person validPerson = new PersonBuilder().build();
+    public void execute_newMeeting_success() {
+        Engagement validMeeting = new EngagementBuilder().buildAsMeeting();
+
+        Model expectedModel = new ModelManager(model.getEngagementList(), new UserPrefs());
+        expectedModel.addEngagement(validMeeting);
+        expectedModel.saveEngagementList();
```

Comment:
I think can extract this line into addEngagement to fix. Can leave it to me :)

Reply:
Ok sure :)